packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -323,6 +331,69 @@ export async function uploadData<T extends UploadDataLookup, K extends keyof T>(
   );
 }
 
+export function shrinkRawData<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
+): Boolean {
+  let totalSize = Buffer.byteLength(JSON.stringify(data));
+
+  while (totalSize > UPLOAD_SIZE_MAX) {
+    let largestEntryKey = '';
+    let largestEntrySize = 0;
+    let largestRawDataEntryKey = '';
+    let largestRawDataEntrySize = 0;
+    let largestItemKey = '';
+    let largestItemSize = 0;
+
+    // Find largest Entity
+    for (const entry in data) {
+      if (JSON.stringify(data[entry]).length > largestEntrySize) {
+        largestEntrySize = JSON.stringify(data[entry]).length;
+        largestEntryKey = entry;
+      }
+    }
+
+    // Find largest _rawData entry (typically 0, but check to be certain)
+    for (const rawEntry in data[largestEntryKey]['_rawData']) {
+      if (
+        JSON.stringify(data[largestEntryKey]['_rawData'][rawEntry]).length >
+        largestRawDataEntrySize
+      ) {
+        largestRawDataEntrySize = JSON.stringify(
+          data[largestEntryKey]['_rawData'][rawEntry],
+        ).length;
+        largestRawDataEntryKey = rawEntry;
+      }
+    }
+
+    // Find largest item within rawData
+    for (const item in data[largestEntryKey]['_rawData'][
+      largestRawDataEntryKey
+    ]['rawData']) {
+      if (
+        data[largestEntryKey]['_rawData'][largestRawDataEntryKey]['rawData'][
+          item
+        ] &&
+        data[largestEntryKey]['_rawData'][largestRawDataEntryKey]['rawData'][
+          item
+        ].length > largestItemSize
```

Comment:
Is the `item` guaranteed to have the `.length` property? 
What if the property is actually an object, is the intent to find the largest nested property or just cut out the entire object? 
We may want to add a few test cases with different data types.
Also, isn't `default` the most common root property within rawData? I could be totally wrong on this one.

Reply:
Good catch!  I'd actually intended to do a JSON.stringify() call on that item so that it would be the closest thing to an apples-to-apples comparison for finding the largest overall thing no matter what the data type.  I'll get that updated and you can have another look and see if you're still concerned with it.

Regarding `default` being the most common root property within rawData, yes I believe you're correct that `default` is the most common value for the `name` property, i.e.

"_rawData": [
        {
          "name": "default",
          "rawData": {
            "admin_id": "...",
            "admin_units": [],
            ...
          }
        }
      ]